src/tracked_struct/struct_map.rs

```diff
@@ -234,6 +286,14 @@ where
     pub fn get<'db>(&'db self, current_revision: Revision, id: Id) -> C::Struct<'db> {
         StructMap::get_from_map(&self.map, current_revision, id)
     }
+
+    pub fn get_and_validate_last_changed<'db>(
```

Comment:
Can we make this method `pub(super)`? It only seems to be called from `tracked_field`. I also suggest that we document the panic conditions.